src/manifolds/Symplectic.jl

```diff
@@ -0,0 +1,889 @@
+"""
+struct RealSymplecticMetric <: RiemannianMetric end
+
+struct ExtendedSymplecticMetric <: AbstractMetric end
+
+struct CayleyInverseRetraction <: AbstractInverseRetractionMethod end
+
+@doc raw"""
+    SymplecticMatrix{T}
+
+A lightweight structure to represent the action of the matrix
+representation of the canonical symplectic form,
+````math
+Q_{2n} =
+\begin{bmatrix}
+0_n & I_n \\
+ -I_n & 0_n
+\end{bmatrix} \quad \in ℝ^{2n \times 2n},
+````
+such that
+````math
+\omega_{2n}(x, y) = x^TQ_{2n}y, \quad x, y \in ℝ^{2n}.
+````
+
+The entire matrix is however not instantiated in memory, instead a scalar
+``λ`` of type `T` is stored, which is used to keep track of operations and multiplications
+applied  to each `SymplecticMatrix`. For example, given `Q = SymplecticMatrix(1.0)`,
+represented as `1.0*[0 I; -I 0]`, the adjoint `Q'` returns `SymplecticMatrix(-1.0)`.
+"""
+struct SymplecticMatrix{T}
+    λ::T
+end
+SymplecticMatrix() = SymplecticMatrix(1)
+SymplecticMatrix(λ::T) where {T<:Number} = SymplecticMatrix{T}(λ)
+
+function SymplecticMatrix(arrays::Vararg{AbstractArray})
+    TS = Base.promote_type(map(eltype, arrays)...)
+    return SymplecticMatrix(one(TS))
+end
+
+@doc raw"""
+    change_representer(::Symplectic, ::EuclideanMetric, p, X)
+    change_representer!(::Symplectic, Y, ::EuclideanMetric, p, X)
+
+Compute the representation of a tangent vector ``χ ∈ T_p\operatorname{Sp}(2n, ℝ)`` s.t.
+````math
+    g_p(c_p(χ), η) = ⟨χ, η⟩^{\text{Euc}} \;∀\; η ∈ T_p\operatorname{Sp}(2n, ℝ).
+````
+with the conversion function
+````math
+    c_p : T_p\operatorname{Sp}(2n, ℝ) \rightarrow T_p\operatorname{Sp}(2n, ℝ), \quad
+    c_p(η) = \frac{1}{2} pp^T η + \frac{1}{2} pQ η^T pQ.
+````
+
+Each of the terms ``c_p^1(η) = p p^T η`` and ``c_p^2(η) = pQ η^T pQ`` from the
+above definition of ``c_p(η)`` are themselves metric compatible in the sense that
+````math
+    c_p^i : T_p\operatorname{Sp}(2n, ℝ) \rightarrow \mathbb{R}^{2n \times 2n}\quad
+    g_p^i(c_p(χ), η) = ⟨χ, η⟩^{\text{Euc}} \;∀\; η ∈ T_p\operatorname{Sp}(2n, ℝ),
+````
+for ``i \in {1, 2}``. However the range of each function alone is not confined to
+``T_p\operatorname{Sp}(2n, ℝ)``, but the convex combination
+````math
+    c_p(η) = \frac{1}{2}c_p^1(η) + \frac{1}{2}c_p^2(η)
+````
+does have the correct range ``T_p\operatorname{Sp}(2n, ℝ)``.
+"""
+change_representer(::Symplectic, ::EuclideanMetric, p, X)
+
+function change_representer!(::Symplectic, Y, ::EuclideanMetric, p, X)
+    Q = SymplecticMatrix(p, X)
+    Y .= (1 / 2) .* p * (p' * X .+ Q * X' * p * Q)
+    return Y
+end
+
+@doc raw"""
+    change_representer(MetMan::MetricManifold{𝔽, Euclidean{Tuple{m, n}, 𝔽}, ExtendedSymplecticMetric},
+                       EucMet::EuclideanMetric, p, X)
+    change_representer!(MetMan::MetricManifold{𝔽, Euclidean{Tuple{m, n}, 𝔽}, ExtendedSymplecticMetric},
+                        Y, EucMet::EuclideanMetric, p, X)
+
+Change the representation of a matrix ``χ ∈ \mathbb{R}^{2n \times 2n}``
```

Comment:
Having the argument named `X` and the matrix `χ` may be a bit misleading, it's very easy to confuse these two symbols.

Reply:
Good point, I'll change to a different letter like zeta or something for the documentation.